index.js

```diff
@@ -1,170 +1,164 @@
+        var currentHoveredEl = document.getElementById('tc-hover');
+        currentHoveredEl && currentHoveredEl.removeAttribute('id');
 
-    bindings(this);
+        el && el.setAttribute('id', 'tc-hover');
+    }
 
-    this.addValues = function (vals) {
-        masterList = dedupe(masterList.concat(vals));
-    };
-}
+    function highlightFocusedOption(options, indexOfCurrentOption) {
+        // Allow for going over and under in list
+        options = options || [];
+        var lengthOfOptions = options.length;
+        if (indexOfCurrentOption >= lengthOfOptions) {
+            indexOfCurrentOption = 0;
+        } else if (indexOfCurrentOption < 0) {
+            indexOfCurrentOption = lengthOfOptions - 1;
+        }
+
+        addActiveClass(options[indexOfCurrentOption]);
+        return indexOfCurrentOption;
+    }
+
+    function navigateListListener(inputEl, id) {
+        var indexOfCurrentHighlightedItem = -1;
+        inputEl.addEventListener('keydown', function (e) {
+            var options = (document.querySelector('.tc-' + id) || {}).children;
+            var keycode = e.keyCode;
+            if (keycode === 40) { // upKey
+                indexOfCurrentHighlightedItem = highlightFocusedOption(options, indexOfCurrentHighlightedItem += 1);
+            } else if (keycode === 38) { // Down Key
+                indexOfCurrentHighlightedItem = highlightFocusedOption(options, indexOfCurrentHighlightedItem -= 1);
+            } else if (keycode === 13) { // Enter Key
+                indexOfCurrentHighlightedItem >= 0 && options.length > 0 && options[indexOfCurrentHighlightedItem].click();
+            }
+        });
+    }
+
+    // Internals
+
+    function TinyComplete(options) {
+        if (typeof options !== 'object') return;
+        var masterList = dedupeLists(options.listItems);
+        var valuesToShow = masterList;
+        var userSearchQuery = '';
+
+        setInputBindings();
+
+        this.addListItems = function (vals) {
+            masterList = dedupeLists(masterList.concat(vals));
+        };
+
+        function setInputBindings() {
+            var inputEl = document.getElementById(options.id);
+            inputEl.addEventListener('input', function (e) {
+                userSearchQuery = e.target.value;
+                valuesToShow = filterResults(masterList, userSearchQuery);
+                addDropDownHTML(inputEl, valuesToShow, options);
+                options.onUserInput ? options.onUserInput(valuesToShow, userSearchQuery) : /* istanbul ignore next */ function () {};
+            });
+            navigateListListener(inputEl, options.id);
+        }
+    }
 
-(function () {
     /* istanbul ignore next  */
     if (typeof module !== 'undefined' && module.exports) {
         module.exports = TinyComplete;
-    } else if (typeof define === 'function' && define.amd) {
+    } else if (typeof window.define === 'function' && define.amd) {
         window.define(TinyComplete);
```

Comment:
this still doesn't quite cover it because now you've checked for `window.define` but then accessed `this.define.amd` and there's no guarantee that `window` and `this` are the same

I think it's probably better to just do this:
```
if (typeof module !== 'undefined' && module.exports) {
         module.exports = TinyComplete;
} else if (typeof define === 'function' && define.amd) {
         define(TinyComplete);
} else if (typeof window !== 'undefined') {
         window.TinyComplete = TinyComplete;
}
```